contrib/install/install.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+#
+#
+#  Copyright 2016 CUBRID Corporation
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+function get_cubrid_dir() {
+  cubrid_str=/CUBRID/
+  cubrid_str_len=${#cubrid_str}
+  current_dir="`pwd`"
+
+  find_dir=$(echo ${current_dir} | grep ${cubrid_str})
+
+  if [ -z "$find_dir" ];then
+    echo ""
+  else
+    temp_dir=${current_dir%%$cubrid_str*}
+    temp_dir_len=${#temp_dir}
+    cubrid_dir_len=`expr $temp_dir_len + $cubrid_str_len`
+    cubrid_dir=$(echo $current_dir | cut -c1-$cubrid_dir_len)
+    echo "$cubrid_dir"
+  fi
+}
+
+
+cubrid_home=$(get_cubrid_dir)
+
+if [ -z "$cubrid_home" ];then
+  cubrid_home="`pwd`"
+fi
+
+echo "Is the CUBRID installed in "$cubrid_home" ? [Yn]:"
+
+read line leftover
+is_installed_dir=TRUE
+
+case ${line} in
+  n* | N*)
+    is_installed_dir=FALSE
+esac
+
+if [ "x${is_installed_dir}x" = "xFALSEx" ];then
+  echo ""
+  echo "Please enter the directory where CUBRID is installed: "
+
+  read input_dir leftover
+  cubrid_home=${input_dir}
+fi
+
+if [ ! -d $cubrid_home ];then
+  echo "$cubrid_home: no such directory"
+  exit
+fi
+
+# environment variables for *csh
+cubrid_csh_envfile="$HOME/.cubrid.csh"
+
+cat << ____cubrid__here_doc____ > ${cubrid_csh_envfile}_temp
+setenv CUBRID ${cubrid_home}
+____cubrid__here_doc____
+cat << '____cubrid__here_doc____' >> ${cubrid_csh_envfile}_temp
+setenv CUBRID_DATABASES $CUBRID/databases
+if (${?LD_LIBRARY_PATH}) then
+  setenv LD_LIBRARY_PATH $CUBRID/lib:${LD_LIBRARY_PATH}
+else
+  setenv LD_LIBRARY_PATH $CUBRID/lib
+endif
+setenv SHLIB_PATH $LD_LIBRARY_PATH
+setenv LIBPATH $LD_LIBRARY_PATH
+set path=($CUBRID/bin $path)
+____cubrid__here_doc____
+
+
+# environment variables for *sh
+cubrid_sh_envfile="$HOME/.cubrid.sh"
+cat << ____cubrid__here_doc____ > ${cubrid_sh_envfile}_temp
+CUBRID=${cubrid_home}
+____cubrid__here_doc____
+cat << '____cubrid__here_doc____' >> ${cubrid_sh_envfile}_temp
+CUBRID_DATABASES=$CUBRID/databases
+if [ "x${LD_LIBRARY_PATH}x" = xx ]; then
+  LD_LIBRARY_PATH=$CUBRID/lib
+else
+  LD_LIBRARY_PATH=$CUBRID/lib:$LD_LIBRARY_PATH
+fi
+SHLIB_PATH=$LD_LIBRARY_PATH
+LIBPATH=$LD_LIBRARY_PATH
+PATH=$CUBRID/bin:$PATH
+export CUBRID
+export CUBRID_DATABASES
+export LD_LIBRARY_PATH
+export SHLIB_PATH
+export LIBPATH
+export PATH
+
+LIB=$CUBRID/lib
```

Comment:
Where is the CUBRID variable set ?
Is it set on line 89 ?

Reply:
The CUBRID variable is set in line 89.

Reply:
sorry, I misunderstand the code.